content/concepts/consensus-network/amendments/known-amendments.md

```diff
@@ -339,6 +340,18 @@ The fix1623 amendment has no effect on [CheckCash transactions][] for a fixed am
 **Caution:** In `rippled` 1.0.0, if the Checks amendment is enabled before the fix1623 amendment, the `delivered_amount` may display as "0" for variable-amount CheckCash transactions from before this amendment was enabled, even if the transaction delivered a nonzero amount. Ripple plans to enable fix1623 at the same time as the [Checks][] amendment on the production network, but this situation may be possible on [parallel networks](parallel-networks.html).
 
 
+## fix1781
+[fix1781]: #fix1781
+
+| Amendment ID                                                     | Status    |
+|:-----------------------------------------------------------------|:----------|
+| 25BA44241B3BD880770BFA4DA21C7180576831855368CBEC6A3154FDE4A7676E | In Development |
+
+Fixes a bug where certain XRP endpoints were not checked when detecting circular paths.
+
+Without this amendment, in some circumstances it may be possible to create and execute "circular" payments where the same account sends and receives the same currency. With this amendment, those payments fail with the [`temBAD_PATH_LOOP` result code](tem-codes.html) instead.
+
```

Comment:
I'd reword this slightly so it talks about payment paths and path steps rather than the "same account". We disallow such payment even if different accounts are involved. Maybe
```
Without this amendment, it was possible to have a payment path where the input to the path
is XRP, and an intermediate path step also output XRP. This is a "circular" payment and it should 
have been disallowed. With this amendment, those payments fail with the
[`temBAD_PATH_LOOP` result code](tem-codes.html) instead.
```

Reply:
Reworded slightly to use present tense and to clarify _why_ that qualifies as a loop:

> Without this amendment, it is possible to have a [payment path](paths.html) where the input to the path is XRP, and an intermediate path step also outputs XRP. This is a "circular" payment, because it could be completed more efficiently with a shorter path that skips everything up to the intermediate step that outputs XRP.

Reply:
I'd leave off the "it could be completed more efficiently", that's not the motivation for the rule.

The reason we disallow payments like this has to do with the inner workings of the payment engine. We first "execute" the path backwards to see how much we need to put in to get the desired amount out, then we "execute" the path forward (because we may not be able to get the desired amount out). If we had circular payments, then one step on the path could defund another step on the path and paths would act differently in the forward direction than in the reverse direction. To keep the code simple, we do not allow such cases. I would not put that in the explanation (that's too far into the weeds), but wanted to explain the rational for the rule.